charts/consul/templates/client-daemonset.yaml

```diff
@@ -509,6 +509,12 @@ spec:
         - name: CONSUL_LOGIN_META
           value: "component=client,pod=$(NAMESPACE)/$(POD_NAME)"
         {{- end }}
+        - name: CONSUL_LOGIN_DATACENTER
```

Comment:
I think you might need bats tests in client-daemonset.bats to check the output of the toggle.

Reply:
Yeah I thought this was covered by https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/test/unit/api-gateway-controller-deployment.bats#L497
However, I think I misunderstood that test, it is for the command and I also need to add one for the env vars. Does that seem correct?

Reply:
Added